lib/_http_outgoing.js

```diff
@@ -311,6 +318,14 @@ function _storeHeader(firstLine, headers) {
   if (state.sentExpect) this._send('');
 }
 
+function fastTest(field, expression, expressionLength) {
+  if (field.length !== expressionLength) return false;
+
+  return expression.test(field);
+}
+
+exports._fastTest = fastTest;
```

Comment:
Actually, thats not a good name. `strcmpi()` would be, or if that is too unixy, `strCmpInsensitive()`. 

Its unfortunate you need to pass in the length, when the length is derivable as `expression.source.length - 2`, does deriving it have a performance impact?